testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oidc/UserInfoTest.java

```diff
@@ -246,6 +262,219 @@ public void testSuccess_postMethod_header_textEntity() throws Exception {
         }
     }
 
+
+    @Test
+    public void testSuccessEncryptedAndSignedResponse() throws Exception {
+        testUserInfoSignatureAndEncryption(org.keycloak.crypto.Algorithm.RS256, JWEConstants.RSA_OAEP, JWEConstants.A256GCM);
+    }
+
+    @Test
+    public void testSuccessEncryptedResponseAlgRSA_OAEPEncA256GCM() throws Exception {
+        testUserInfoSignatureAndEncryption(null, JWEConstants.RSA_OAEP, JWEConstants.A256GCM);
```

Comment:
18 of 19 tests set null to "alg" of signing. I think most use cases use signed  and encrypt. It seems good to set some valid signature algorithm to "alg" of signing in 18 of 19 tests and null to "alg" of signing in remaining test.

Reply:
+1 to this point

Also regarding test coverage, it is great that your PR has good test coverage. On the other hand, 19 new test methods in the UserInfoTest looks like quite a lot to me. Every new test method adds some overhead when running the automated tests... Is it possible to slightly limit this for example to 5-7 methods instead of 19? Ideally including the possibilities for "null" values correctly resolved to expected defaults as Takashi suggested. IMO it is not strictly needed to test every possible combination, but for example just some "representative" subset of them. WDYT?